### DISCUSSION TITLE
Any Template/Example/Help with a standalone project using a custom Platform?

### DISCUSSION THREAD

**Discussion by yanshay** [slint-ui/slint#3901] [Q&A] [answered]
I developed a Platform for my board based on ESP32, starting from the S3_BOX example and it's working. I managed to get the Carousel example working with it on my board.
However, it was all done under the directory structure of the slint repo, and coupled to slint's build process and local source. And this build and folder hierarchy isn't exactly that simple.

After quite some work I managed to get my own separate project working, but I'm not sure I'm following any best practices for supporting multi platforms and simulator, etc.

Is there an example project of a standalone project with a Platform w/o the entire slint directory structure? Or maybe there's documentation how to do it properly? It feels like I'm reinventing the wheel doing this.

**Comment by ogoffart** [comment, marked as answer]
Are you using Rust or C++?

If using Rust, there is this template: https://github.com/slint-ui/slint-mcu-rust-template
If C++, the esp-idf examples from this repository can just be taken out as the directories are self contained.

**Reply by yanshay** [reply]
Thanks, I'm using Rust. This is what I was looking for.
I'd just say my 2 cents - I think separating the mcu dependent code to a separate file/modules, similar to how it was done on the slint examples folder, would be beneficial. 
So if I for example want to add a specific board ESP32 support, it would be just about adding stuff and then building with relevant features rather than have to move code around to add my mcu to the project (I do want to benefit from having the pico supported as well with my project if the code exists).
And then it would be possible for contributors to add support for boards, and expand the reach of slint to more and more boards with time.
